src/query/functions/src/aggregates/aggregate_sum.rs

```diff
@@ -99,12 +99,24 @@ where
     }
 
     fn accumulate_keys(places: &[StateAddr], offset: usize, columns: &Column) -> Result<()> {
-        let darray = NumberType::<T>::try_downcast_column(columns).unwrap();
-        darray.iter().zip(places.iter()).for_each(|(c, place)| {
-            let place = place.next(offset);
-            let state = place.get::<Self>();
-            state.value += c.as_();
-        });
+        match columns {
+            Column::Nullable(column) => {
```

Comment:
There is no nullable column here, it's wrapped by `NullUnaryAdaptor`.

Reply:
`sum(c)` is always nullable now, because it's wrapped by `OrNullAdaptor`.

Reply:
we can change the return type of eager aggregation function to not null when applying the eager aggregation rule, and then the `ColumnBuilder` will build a non-null column.
https://github.com/datafuselabs/databend/blob/c4786b68b43409776e91cdc7c1626583c21689a0/src/query/service/src/pipelines/processors/transforms/aggregator/transform_aggregate_final.rs#L202-L210